### ISSUE TITLE
input->get_request_header('something'); not working without calling input->request_headers() first

### ISSUE THREAD

**Issue by paranic** [bcit-ci/CodeIgniter#4478]
in an non apache environment.

**not working**
```
$this->input->get_request_header('Host');
$this->input->get_request_header('User-Agent');
```

**working**
```
$this->input->request_headers();
$this->input->get_request_header('Host');
$this->input->get_request_header('User-Agent');
```

also i think ```get_request_header()``` is a little slacky written.

**Comment by narfbg** [comment]
Duplicate of #4056, which was fixed in 3.0.2.  
The current latest release is 3.0.4 and you should even be testing against dev code before reporting a bug.

**Comment by paranic** [comment]
I'm sorry
